### PR TITLE
Limit the size of arg1 to 16 KB

### PR DESCRIPTION
**Pull request by junchaowu**
Limit the size of arg1 to 16KB that matches protocol.

### CODE REVIEW

File: tchannel/tornado/dispatch.py [hunk 1]
```diff
@@ -194,7 +190,7 @@ def handle_call(self, request, connection):
                 # stack context.
                 # The right way to do it is:
                 # with request_context(..):
-                #    future = f()
```
Comment (blampe) [inline tchannel/tornado/dispatch.py]: Hm?

**Comment by blampe** [comment]
Tested / confirmed locally?

File: tchannel/tornado/peer.py [hunk 2]
```diff
@@ -336,6 +332,18 @@ def send(
         arg1.close()
         endpoint = yield read_full(arg1)
 
+        if len(endpoint) > MAX_SIZE_OF_ARG1:
```
Comment (blampe) [inline tchannel/tornado/peer.py]: Why is this in 2 different spots?

File: tests/test_tchannel.py [hunk 3]
```diff
@@ -376,3 +381,18 @@ def test_listen_duplicate_ports():
...
+
+
+@pytest.mark.gen_test
+def test_arg1_limit():
+    tchannel = TChannel(name='client')
+    with pytest.raises(BadRequestError) as e:
+        yield tchannel.call(
+            scheme=schemes.RAW,
+            service='server',
+            arg1='a'*MAX_SIZE_OF_ARG1+'a',
```
Comment (blampe) [inline tests/test_tchannel.py]: Spaces around binary operators.

File: tchannel/tornado/dispatch.py [hunk 4]
```diff
@@ -124,11 +125,18 @@ def handle_call(self, request, connection):
...
-            request.endpoint += chunk
-            chunk = yield request.argstreams[0].read()
+
+        # we have defined the endpoint will fit into the arg1
+        request.endpoint = yield read_full(request.argstreams[0])
+        arg1_size = len(request.endpoint)
+        if arg1_size > MAX_SIZE_OF_ARG1:
+            connection.send_error(
+                BadRequestError(
+                    'arg1 size is %d which exceeds the max size 16KB.' %
```
Comment (abhinav) [inline tchannel/tornado/dispatch.py]: > arg1 is %d bytes long. It must be at most 16kb in length.

**Comment by abhinav** [comment]
Fix error message. Then :shipit:

**Pull request merged**